src/coursescheduler/constraints.py

```diff
@@ -1,2 +1,164 @@
+            if course not in assignment:
+                continue
+            prof = assignment[course]
+            teaching_loads_dict[prof] += 1
+
+        for prof, teachingLoad in teaching_loads_dict.items():
+            if teachingLoad > temp_profs[prof]["teachingObligations"]:
+                return False
+
+        return True
+
+# TODO: Create timeslots for the courses
+# TODO: Create the hard constraints for:
+#  overlapping courses
+#  no-double booked professor
+
+# Hard constraint: the specified courses may not be assigned to overlapping timeslots.
+# class CourseTimeslotOverlapConstraint(Constraint):
+#     def __init__(self, courses) -> None:
+#         super().__init__(courses)
+#         self.courses_ = courses
+#
+#     def satisfied(self, assignment) -> bool:
+#         # Obtain all timeslots for the courses to which this constraint applies.
+#         timeslots: List[TimeSlot] = []
+#         for course in self.courses_:
+#             if course not in assignment:
+#                 continue
+#             for timeslot in assignment[course].timeslots_:
+#                 timeslots.append(timeslot)
+#
+#         # Check for conflicts.
+#         for i in range(len(timeslots)):
+#             for j in range(i + 1, len(timeslots)):
+#                 if (timeslots[i].day_ == timeslots[j].day_):
+#                     if (timeslots[i].start_ >= timeslots[j].start_ and timeslots[i].start_ <= timeslots[j].end_):
+#                         return False
+#                     if (timeslots[j].start_ >= timeslots[i].start_ and timeslots[j].start_ <= timeslots[i].end_):
+#                         return False
+#         return True
+
+
+# # Soft constraint: instructors should only be scheduled to teach when they are available.
+# # Creates a score indicating how much assigned teaching time falls outside of professor's preferred hours.
+# # If this score exceeds a threshold, then the constraint is violated.
+# class InstructorAvailabilityConstraint(Constraint[Course, TimeSlotConfiguration]):
+#     def __init__(self, courses: Course) -> None:
+#         super().__init__(courses)
+#         self.courses_ = courses
+#
+#     def satisfied(self, assignment: Dict[Course, TimeSlotConfiguration]) -> bool:
+#         score = 0
+#         for course in self.courses_:
+#             if course not in assignment:
+#                 continue
+#             for timeslot in assignment[course].timeslots_:
+#                 for preferred_time in course.instructor_.preferred_times_:
+#                     if (timeslot.day_ == preferred_time.day_):
+#                         if (timeslot.start_ < preferred_time.start_):
+#                             score -= (preferred_time.start_ - timeslot.start_) ** 2
+#                         if (preferred_time.end_ < timeslot.end_):
+#                             score -= (timeslot.end_ - preferred_time.end_) ** 2
+#         if (score < -400):
+#             return False
+#
+#         return True
```

Comment:
It may be worth removing the commented soft constraints for now and leaving them for #35 and #14 as they may not even be used.

```suggestion
```